src/commands/contract.ts

```diff
@@ -0,0 +1,205 @@
+import { Command } from "commander";
+import fs from "fs/promises";
+import path from "path";
+import { exec, spawn } from "child_process";
+import { promisify } from "util";
+import inquirer from "inquirer";
+
+const execAsync = promisify(exec);
+
+export const contractCommand = new Command()
+  .name("contract")
+  .description("Generate an AI agent from a Near Protocol contract")
+  .action(async () => {
+    try {
+      const answers = await promptQuestions();
+      const outputDir = path.resolve(answers.output);
+      await fs.mkdir(outputDir, { recursive: true });
+      await generateTypes(outputDir, answers.contract);
+      const code = await generateAIAgent(answers, outputDir);
+      await writeFiles(outputDir, code, answers.contract);
+      await setupAndRunAgent(outputDir);
+
+      console.log("AI agent execution completed successfully.");
+      console.log(`AI agent generated successfully in ${answers.output}`);
+    } catch (error) {
+      console.error("Error generating AI agent");
+      console.error("Error details:", error);
+    }
+  });
+
+
+function showLoadingMessage(message: string): void {
+  console.log(`${message}...`);
+}
+
+async function promptQuestions() {
+  const questions = [
+    {
+      type: 'input',
+      name: 'contract',
+      message: 'Enter the Near Protocol contract name:',
+      validate: (input: string) => input.length > 0 || 'Contract name is required'
+    },
+    {
+      type: 'input',
+      name: 'description',
+      message: 'Enter the contract description (agent instructions):',
+      validate: (input: string) => input.length > 0 || 'Contract description is required'
+    },
+    {
+      type: 'input',
+      name: 'output',
+      message: 'Enter the output directory (press Enter for current directory):',
+      default: '.',
+    },
+    {
+      type: 'input',
+      name: 'accountId',
+      message: 'Enter your near account ID to generate an API key:',
+      validate: (input: string) => input.length > 0 || 'Near account ID is required'
+    }
+  ];
+
+  return await inquirer.prompt<{
+    contract: string;
+    description: string;
+    output: string;
+    accountId: string;
+  }>(questions as any);
+}
+
+async function generateTypes(outputDir: string, contract: string) {
+  process.chdir(outputDir);
+  showLoadingMessage("Generating types");
+  await execAsync(`npx near2ts ${contract}`);
+}
+
+async function generateAIAgent(answers: {
+  contract: string;
+  description: string;
+  accountId: string;
+}, outputDir: string) {
+  const apiUrl = "https://contract-to-agent.vercel.app/api/generate";
+
```

Comment:
We will want to prevent abuse from this route

Reply:
Yes, needs some rate limiting